flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/functions/ImplicitConversionEqualsFunctionITCase.java

```diff
@@ -0,0 +1,227 @@
+                        .right(DOUBLE(), DOUBLE_DATA)
+                        .right(DECIMAL(1, 0), DECIMAL_DATA)
+                        .build(),
+                TypeConversionTestBuilder.left(FLOAT(), FLOAT_DATA)
+                        .right(FLOAT(), FLOAT_DATA)
+                        .right(DOUBLE(), DOUBLE_DATA)
+                        .right(DECIMAL(1, 0), DECIMAL_DATA)
+                        .build(),
+                TypeConversionTestBuilder.left(DOUBLE(), DOUBLE_DATA)
+                        .right(DOUBLE(), DOUBLE_DATA)
+                        .right(DECIMAL(1, 0), DECIMAL_DATA)
+                        .build());
+    }
+
+    private static List<TestSpec> implicitConversionBetweenTimeAndString() {
+        return Arrays.asList(
+                TypeConversionTestBuilder.left(DATE(), DATE_DATA)
+                        .right(DATE(), DATE_DATA)
+                        .right(STRING(), STRING_DATA_EQUALS_DATE)
+                        .build(),
+                TypeConversionTestBuilder.left(TIME(), TIME_DATA)
+                        .right(STRING(), STRING_DATA_EQUALS_TIME)
+                        .build(),
+                TypeConversionTestBuilder.left(TIMESTAMP(), TIMESTAMP_DATA)
+                        .right(STRING(), STRING_DATA_EQUALS_TIMESTAMP)
+                        .build());
+    }
+
+    // unsupported temporarily
+    private static List<TestSpec> unsupportedImplicitConversionBetweenNumericAndString() {
+        return Arrays.asList(
+                TypeConversionTestBuilder.left(STRING(), STRING_DATA_EQUALS_NUMERIC)
+                        .right(STRING(), STRING_DATA_EQUALS_NUMERIC)
+                        .fail(TINYINT(), TINY_INT_DATA)
+                        .fail(SMALLINT(), SMALL_INT_DATA)
+                        .fail(INT(), INT_DATA)
+                        .fail(BIGINT(), BIG_INT_DATA)
+                        .fail(FLOAT(), FLOAT_DATA)
+                        .fail(DOUBLE(), DOUBLE_DATA)
+                        .fail(DECIMAL(1, 0), DECIMAL_DATA)
+                        .build());
+    }
+
+    static class TypeConversionTestBuilder {
+        private TestSpec testSpec;
+        private DataType leftType;
+        private final List<Object> columnData = new ArrayList<>();
+        private final List<DataType> columnTypes = new ArrayList<>();
+        private final List<Boolean> isSuccessResults = new ArrayList<>();
+
+        private static TypeConversionTestBuilder left(DataType leftType, Object leftValue) {
+            TypeConversionTestBuilder builder = new TypeConversionTestBuilder();
+            builder.leftType = leftType;
+            builder.columnData.add(leftValue);
+            builder.columnTypes.add(leftType);
+            builder.testSpec =
+                    TestSpec.forFunction(
+                            BuiltInFunctionDefinitions.EQUALS,
+                            "The left type in type conversion is " + leftType.toString());
```

Comment:
You could simplify the description, like: `"left: " + leftType.toString()`

Reply:
got it.